api/src/main/java/io/cloudevents/CloudEventAttributes.java

```diff
@@ -78,9 +79,25 @@
     @Nullable
     ZonedDateTime getTime();
 
+    /**
+     * Get the <a href="https://github.com/cloudevents/spec/blob/v1.0/spec.md#context-attributes">context attribute</a> named {@code attributeName}
+     *
+     * @param attributeName a valid attribute name
+     * @return the attribute value or null if this instance doesn't contain such attribute
+     * @throws IllegalArgumentException if the provided attribute name it's not a valid attribute for this spec
+     */
     @Nullable
-    Object getAttribute(String extensionName);
+    Object getAttribute(String attributeName) throws IllegalArgumentException;
 
-    Set<String> getAttributeNames();
+    /**
+     * @return The non-null <a href="https://github.com/cloudevents/spec/blob/v1.0/spec.md#context-attributes">context attributes</a> names in this instance
+     */
+    default Set<String> getAttributeNames() {
```

Comment:
do we even need this method then? I'd propose to drop it , since `getSpecVersion().getAllAttributes()` can be used instead

Reply:
I have no strong opinion on this, i think it can be useful for a user that wants to get only the attributes available in this instance, and it doesn't cost a lot to maintain it